rustls/src/server/tls13.rs

```diff
@@ -715,6 +747,43 @@ mod client_hello {
         common.send_msg(c, true);
     }
 
+    fn emit_compressed_certificate_tls13(
+        transcript: &mut HandshakeHash,
+        common: &mut CommonState,
+        cert_chain: &[CertificateDer<'static>],
+        ocsp_response: Option<&[u8]>,
+        cert_compressor: &'static dyn CertCompressor,
+    ) {
+        let encoding =
+            CertificatePayloadTls13::new(cert_chain.iter(), ocsp_response).get_encoding();
+        let uncompressed_len = encoding.len() as u32;
+
+        let compressed = match cert_compressor.compress(encoding, CompressionLevel::Interactive) {
+            Ok(compressed) => PayloadU24(Payload::new(compressed)),
+            Err(_) => {
+                return emit_certificate_tls13(transcript, common, cert_chain, ocsp_response);
+            }
```

Comment:
This fall-back to uncompressed certs seems interesting. I think it's a valid strategy based on 8879 indicating that peers _MAY_ compress when support is negotiated :+1: I don't see any text in the RFC that discusses this case concretely to offer any counter advice.